### PR TITLE
vimPlugins.fruzzy: init at 2019-10-28

### PR DESCRIPTION
**Pull request by NickHu**
###### Motivation for this change

This adds the vim plugin [fruzzy](https://github.com/raghur/fruzzy/). Note that this plugin has a 'native' component (fuzzy searcher implemented in Nim), so I bootstrapped the Nim dependencies in a little bit of a hackey way (could be made neater if https://github.com/NixOS/nixpkgs/pull/67878 ever gets merged).

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NickHu** [comment]
@GrahamcOfBorg build vimPlugins.fruzzy

File: pkgs/misc/vim-plugins/overrides.nix [hunk 1]
```diff
@@ -214,6 +217,28 @@ self: super: {
     dependencies = with super; [ super.self ];
   });
 
+  fruzzy = let # until https://github.com/NixOS/nixpkgs/pull/67878 is merged, there's no better way to install nim libraries with nix
+    nimpy = fetchTarball "https://github.com/yglukhov/nimpy/archive/master.tar.gz";
+    binaryheap = fetchTarball "https://github.com/bluenote10/nim-heap/archive/master.tar.gz";
```
Comment (jonringer) [inline pkgs/misc/vim-plugins/overrides.nix]: this only works locally, you need to use `fetchurl` + sha256

```
access to URI 'https://github.com/yglukhov/nimpy/archive/master.tar.gz' is forbidden in restricted mode
```

**Comment by NickHu** [comment]
rebased and resolved 'forbidden in restricted mode'
@GrahamcOfBorg build vimPlugins.fruzzy

File: pkgs/misc/vim-plugins/overrides.nix (hunk 1 above)
Comment (NickHu) [reply pkgs/misc/vim-plugins/overrides.nix]: I fixed it by providing sha256 sums as you suggested

**Comment by NickHu** [comment]
@jonringer Have been using this happily for months now, is anything blocking merge? Thanks!

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 81440` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- vimPlugins.fruzzy
</details>

**Comment by jonringer** [comment]
I took a break while my server was being repaired. sorry

**Pull request merged**